syntax = "proto3";

package common;


message SLPoint {
  double s = 1;
  double l = 2;
}

message FrenetFramePoint {
  double s = 1;
  double l = 2;
  double dl = 3;
  double ddl = 4;
}

message SpeedPoint {
  double s = 1;
  double t = 2;
  // speed (m/s)
  double v = 3;
  // acceleration (m/s^2)
  double a = 4;
  // jerk (m/s^3)
  double da = 5;
}

message PathPoint {
  // coordinates
  double x = 1;
  double y = 2;

  // direction on the x-y plane
  double theta = 3;
  // curvature on the x-y planning
  double kappa = 4;
  // accumulated distance from beginning of the path
  double s = 5;

  // derivative of kappa w.r.t s.
  double dkappa = 6;
  // derivative of derivative of kappa w.r.t s.
  double ddkappa = 7;

}

message Path {
  string name = 1;
  repeated PathPoint path_point = 2;
}

message TrajectoryPoint {
  // path point
  PathPoint path_point = 1;
  // linear velocity
  double v = 2;  // in [m/s]
  // linear acceleration
  double a = 3;
  // relative time from beginning of the trajectory
  double relative_time = 4;
  // longitudinal jerk
  double da = 5;
  // The angle between vehicle front wheel and vehicle longitudinal axis
  double steer = 6;
}

message Trajectory {
  repeated TrajectoryPoint trajectory_point = 1;
}


